### ISSUE TITLE
nhmmer reports Alphabet mismatch

### ISSUE THREAD

**Issue by fbosnic** [althonos/pyhmmer#12]
I'm trying to run nhmmer function but can't seem to get the alphabet right. Running
 ```python
    import pyhmmer    
    seq1 = pyhmmer.easel.TextSequence(name=b"seq1", sequence="ACCGACA")
    seq2 = pyhmmer.easel.TextSequence(name=b"seq2", sequence="GGGCCAACA")
    rna = pyhmmer.easel.Alphabet.rna()
    dig1, dig2 = [s.digitize(rna) for s in [seq1, seq2]]
    builder = pyhmmer.plan7.Builder(rna, prior_scheme="alphabet")
    
    gen = pyhmmer.hmmer.nhmmer([dig1], [dig2], builder=builder)
    next(gen)
```

results in
```python
    Traceback (most recent call last):
      File "pyhmmer_test.py", line 10, in <module>
        next(gen)
      File "/apps/conda/fbosnic/envs/test/lib/python3.8/site-packages/pyhmmer/hmmer.py", line 310, in _multi_threaded
        raise thread.error from None
      File "/apps/conda/fbosnic/envs/test/lib/python3.8/site-packages/pyhmmer/hmmer.py", line 112, in run
        self.process(index, query)
      File "/apps/conda/fbosnic/envs/test/lib/python3.8/site-packages/pyhmmer/hmmer.py", line 125, in process
        hits = self.search(query)
      File "/apps/conda/fbosnic/envs/test/lib/python3.8/site-packages/pyhmmer/hmmer.py", line 166, in search
        return self.pipeline.search_seq(query, self.sequences, self.builder)
      File "pyhmmer/plan7.pyx", line 3777, in pyhmmer.plan7.Pipeline.search_seq
      File "pyhmmer/plan7.pyx", line 3819, in pyhmmer.plan7.Pipeline.search_seq
    pyhmmer.errors.AlphabetMismatch: Expected Alphabet.amino(), found Alphabet.rna()
```
Am I using it correctly, does the alphabet need to be set somewhere else as well?

As far as I could trace it, the following line might be the cause
https://github.com/althonos/pyhmmer/blob/eb6fe7c0e74557e0ae9d647693711583d2d86b68/pyhmmer/hmmer.py#L82

**Comment by fbosnic** [comment]
Hi @althonos,

I see, thanks for the answer. I would be very interested in using the function once this is sorted out.

**Comment by althonos** [comment]
Hi @fbosnic ,

The difference between `nhmmer` and `phmmer` is that `nhmmer` will use the "long target pipeline" instead of the default pipeline used by everything else. This allows pipelined searches to run on target sequences longer than 100,000 residues using a sliding window to process each target.

I've released `v0.4.9`, which has a fixed implementation for `nhmmer`. The function now accepts `HMM` arguments in addition to `DigitalSequence` and `DigitalMSA` arguments, and will run a long target search pipeline. I have added tests to make sure the results are consistent between HMMER and PyHMMER on the same query HMM and target genome.
